src/xcode/ENA/ENA/Source/AppDelegate & Globals/Route.swift

```diff
@@ -47,33 +47,36 @@ enum Route {
 				Log.error("Antigen test data is nil, either timeStamp is -ve or the hash is invalid", log: .qrCode)
 				return
 			}
-			let allIsNil = testInformation.firstName == nil && testInformation.lastName == nil && testInformation.dateOfBirthString == nil
-			// dateOfBirth is nil if dateOfBirthString is nil OR if dateOfBirthString is invalid
-			let allIsValid = testInformation.firstName != nil && testInformation.lastName != nil && testInformation.dateOfBirth != nil
 			
-			guard allIsNil || allIsValid else {
-				self = .rapidAntigen( .failure(.invalidTestCode(.invalidTestedPersonInformation)))
-				Log.error("Antigen test data is not valid: all values are nil? \(allIsNil), all values are valid? \(allIsValid)", log: .qrCode)
-				return
-			}
+			let firstName = testInformation.firstName ?? ""
+			let lastName = testInformation.lastName ?? ""
+			let dateOfBirthString = testInformation.dateOfBirthString ?? ""
+			let salt = testInformation.cryptographicSalt ?? ""
+			let timestamp = String(testInformation.timestamp)
+			let recomputedHashString: String
 			
-			if allIsValid {
+			if firstName.isEmpty && lastName.isEmpty && dateOfBirthString.isEmpty {
```

Comment:
From the tech spec: "dob` (date of birth) must have the format `YYYY-MM-DD`"
I'm missing it here. Do I have Tomaten auf den Augen?

Reply:
This is not part of the ticket or this PR